Use child window

AugusteBa opened the pull request.

AugusteBa commented:
@oriondean, could you have a look at this PR to see if anything should be changed? I'll update readme tomorrow.

oriondean commented on demo/src/pages/childWindow/ChildWindow.tsx:
```diff
@@ -0,0 +1,70 @@
+                This hook returns allows creating and managing child windows.
+            </div>
+            <div className={styles.description}>
+                Try clikcing on a button to see a childWindow launched.
```
clicking*

oriondean commented on demo/src/pages/childWindow/ChildWindow.tsx:
```diff
@@ -0,0 +1,70 @@
+        <div className={styles.container}>
+            <h1 className={styles.title}>useChildWindow</h1>
+            <div className={styles.description}>
+                This hook returns allows creating and managing child windows.
```
Would something like `This hook allows you to create and manage a single child window` be better wording here?

oriondean commented on demo/src/pages/childWindow/ChildWindow.tsx:
```diff
@@ -0,0 +1,70 @@
+
+import styles from "./ChildWindow.module.css";
+
+const codeExample = `import {useBounds} from "openfin-react-hooks";
```
This example needs updating

oriondean commented on index.d.ts:
```diff
@@ -28,6 +29,14 @@ export interface IChannelAction {
   action: Action;
 }
 
+export interface IUseChildWindow {
```
IUseChildWindowOptions?

oriondean commented on package.json:
```diff
@@ -39,6 +39,8 @@
   },
   "dependencies": {
-    "openfin-layouts": "^1.0.3"
+    "@types/react-dom": "^16.9.1",
```
`@types` should be in devDependencies

oriondean commented on src/useChildWindow.ts:
```diff
@@ -0,0 +1,147 @@
+    [injectNode],
+  );
+
+  const inheritFromParent = useCallback(() => {
```
Could this behaviour be made optional?

oriondean commented on src/useChildWindow.ts:
```diff
@@ -0,0 +1,147 @@
+
+      if (cssUrl) {
+        const linkElement = childWindow.window
+        .getWebWindow().document.createElement("link");
```
indent

AugusteBa commented:
@oriondean, I addressed your comments, thanks for having a look. Could you look through my changes again, just to make sure that i have got everything right?

oriondean commented on demo/src/pages/childWindow/interfaces/ILaunchConfig.tsx:
```diff
@@ -0,0 +1,11 @@
+interface ILaunchConfig {
```
Could this file just be a definition file instead? e.g. `ILaunchConfig.d.ts` + `export interface ILaunchConfig`

oriondean commented on src/useChildWindow.ts:
```diff
@@ -0,0 +1,181 @@
+
+  useEffect(() => {
+    if (childWindow.state === CHILD_WINDOW_STATE.LAUNCHED) {
+      if (jsx) {
```
Can combine these two if statements

oriondean commented on demo/src/pages/home/Home.tsx:
```diff
@@ -88,6 +88,13 @@ const cards: IDemoCard[] = [
       name: "useChannels",
     },
+    {
+      description: "Launch and manage child windows",
```
Create and manage a single child window*?
